diff --git a/bfd/peXXigen.c b/bfd/peXXigen.c
index 13e39e4..deef7b3 100644
--- a/bfd/peXXigen.c
+++ b/bfd/peXXigen.c
@@ -873,6 +873,8 @@ _bfd_XXi_only_swap_filehdr_out (bfd * abfd, void * in, void * out)
   /* Only use a real timestamp if the option was chosen.  */
   if ((pe_data (abfd)->insert_timestamp))
     H_PUT_32 (abfd, time (0), filehdr_out->f_timdat);
+  else
+    H_PUT_32 (abfd, 0, filehdr_out->f_timdat);

   PUT_FILEHDR_SYMPTR (abfd, filehdr_in->f_symptr,
                    filehdr_out->f_symptr);
